File: tuts/010-cloudmap-service-discovery/dotnet/CloudMapWrapper.cs
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

using Amazon.CloudMap;
using Microsoft.Extensions.Logging;

public class CloudMapWrapper
{
    private readonly IAmazonCloudMap _client;
    private readonly ILogger<CloudMapWrapper> _logger;

    public CloudMapWrapper(IAmazonCloudMap client, ILogger<CloudMapWrapper> logger)
    {
        _client = client;
        _logger = logger;
    }

    // TODO: Add async wrapper methods matching CLI tutorial actions
}

File: tuts/010-cloudmap-service-discovery/php/CloudMapWrapper.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

namespace Example\CloudMap;

use Aws\CloudMap\CloudMapClient;
use Aws\Exception\AwsException;

class CloudMapWrapper
{
    private CloudMapClient $client;

    public function __construct(CloudMapClient $client)
    {
        $this->client = $client;
    }

    // TODO: Add wrapper methods matching CLI tutorial actions
}
